### ISSUE TITLE
NoMethodError: private method `importmap=' called

### ISSUE THREAD

**Issue by javierav** [rails/importmap-rails#47]
I'm updating `importmap-rails` from 0.6.1 to 0.7.6 and I get the following error:

```
NoMethodError: private method `importmap=' called for #<XXXX::Application:0x000056001ecded98>
/home/runner/work/xxxx/xxxx/vendor/bundle/ruby/2.6.0/gems/importmap-rails-0.7.6/lib/importmap/engine.rb:14:in `block in <class:Engine>'
```

* importmap-rails 0.7.6
* stimulus-rails 0.6.0
* turbo-rails 0.8.1
* rails 6.1.4.1

```
# config/importmap.rb

pin 'bootstrap', to: 'https://cdn.skypack.dev/bootstrap'
pin '@hotwired/turbo-rails', to: 'turbo.js'
pin '@hotwired/stimulus', to: 'stimulus.js'
pin '@hotwired/stimulus-autoloader', to: 'stimulus-autoloader.js'
pin '@hotwired/stimulus-importmap-autoloader', to: 'stimulus-importmap-autoloader.js'
pin '@rails/activestorage', to: 'https://cdn.skypack.dev/@rails/activestorage'
pin '@rails/actioncable', to: 'https://cdn.skypack.dev/@rails/actioncable'
pin 'trix', to: 'https://cdn.skypack.dev/trix'
pin '@rails/actiontext', to: 'https://cdn.skypack.dev/@rails/actiontext'

pin_all_from 'app/assets/javascripts/channels', under: 'channels'
pin_all_from 'app/assets/javascripts/controllers', under: 'controllers'

pin 'application'
```

**Comment by dhh** [comment]
You're calling `Rails.application.config.importmap` somewhere and that needs to be changed to `Rails.application.importmap`.

**Comment by javierav** [comment]
My App does not have any reference to `Rails.application.config.importmap`:

```bash
~/app $ grep -nir 'config.importmap' *
~/app $
```

The failure happen in this point of `importmap-rails/lib/importmap/engine.rb`:

```ruby
initializer "importmap" do |app|
  app.importmap = Importmap::Map.new.draw("config/importmap.rb")
end
```

For some reason `app.importmap` and `app.importmap=` are private methods!

**Comment by dhh** [comment]
Do you have stimulus-rails or turbo-rails installed? If so, make sure you're running the latest versions.

**Comment by javierav** [comment]
@dhh Yes, I'm using the latest available versions...

```
importmap-rails 0.7.6
stimulus-rails 0.6.0
turbo-rails 0.8.1
rails 6.1.4.1
```

**Comment by dhh** [comment]
Is importmap-rails listed before the other two gems in your Gemfile?

**Comment by javierav** [comment]
> Is importmap-rails listed before the other two gems in your Gemfile?

Yes, it's listed before!

> Can't replicate when I do:

I can replicate it following your installation steps:

`rails new sport` -> OK
`cd sport`-> OK
`bundle add importmap-rails` -> OK
`bin/rails importmap:install` -> FAILURE

```
$ bin/rails importmap:install
rails aborted!
NoMethodError: private method `importmap=' called for #<Sport::Application:0x00007f97bdf9f2c0>
/Users/javierav/Desarrollo/sport/config/environment.rb:5:in `<main>'
/Users/javierav/Desarrollo/sport/bin/rails:5:in `<top (required)>'
/Users/javierav/Desarrollo/sport/bin/spring:10:in `require'
/Users/javierav/Desarrollo/sport/bin/spring:10:in `block in <top (required)>'
/Users/javierav/Desarrollo/sport/bin/spring:7:in `tap'
/Users/javierav/Desarrollo/sport/bin/spring:7:in `<top (required)>'
Tasks: TOP => app:template => environment
(See full trace by running task with --trace)
```

I'm using MRI Ruby: `ruby 2.6.7p197 (2021-04-05 revision 67941) [x86_64-darwin20]`

**Comment by dhh** [comment]
Weird. Try Ruby 2.7.

**Comment by javierav** [comment]
`ruby 2.6.7p197 (2021-04-05 revision 67941) [x86_64-darwin20]` -> FAILURE
`ruby 2.6.8p205 (2021-07-07 revision 67951) [x86_64-darwin20]` -> FAILURE
`ruby 2.7.4p191 (2021-07-07 revision a21a3b7d23) [x86_64-darwin20]` -> OK

Any problem with Ruby 2.6.x and `Rails::Application.send(:attr_accessor, :importmap)`? 🤔

**Comment by terencechow** [comment]
I know this is closed but I just had the same issue with a jruby app and sprockets-rails. Jruby 3 follows ruby 2.6.  Specifically I got 

```
NoMethodError:
    private method `assets=` called for #<RailsRoot::Application:0x47e7b7>
   # ...'path/to/sprockets-rails/lib/sprockets/railtie.rb:209
```

Inside sprockets I see assets defined with `attr_accessor :assets` which is the same functionality used in this library.

**Comment by headius** [comment]
JRuby 9.4 will be out soon with support for Ruby 3.1 and Rails 7.

**Comment by headius** [comment]
This issue is fixed on JRuby master (9.4 and will not be an issue at release.
